pkgs/os-specific/linux/dcgm/default.nix

```diff
@@ -0,0 +1,113 @@
+{ lib
+, callPackage
+, gcc11Stdenv
+, fetchFromGitHub
+, catch2
+, cmake
+, cudaPackages_10_2
+, cudaPackages_11_8
+, cudaPackages_12
+, fmt
+, git
+, jsoncpp
+, libevent
+, libyamlcpp
+, plog
+, python3
+, tclap_1_4
+
+# Allow builds targeting non-NixOS distros to adjust where libcuda.so is found
```

Comment:
Do you have a practical example of this and why using `addOpenGLRunpath` directly wouldn't be enough?

Because afaik other distros rely on ldconfig and the FHS to locate libraries, not the library rpath.

Reply:
I am deploying this package to both NixOS machines and RHEL machines with Nix installed. On the RHEL machines, it's important that I have control of the *order* in which libraries get loaded. I need libcuda to be loaded from `/lib[64]` but I don't want anything else loaded from there. So this option allows the Nix glibc etc. to take precedence over any FHS paths. I can easily override this option to make a build variant that works on RHEL.

Reply:
I'm honestly not sure if you can use the RPATH to prevent that behavior since it only affects the load order for that specific dynamic library. Once you load something outside of the nix store it can load anything it wants. The only way to prevent this is to load the required libraries yourself before you load the external library.

Reply:
For so files yes, but for bins the RPATH is reliable unless `LD_*` things are in the mix... but in that scenario it's the caller's fault. ;)

Reply:
That honestly sounds better since `addOpenGLRunpath` should be enough for NixOS and gives a single parameter to override for your specific use-case.